From 9300034c578799df622f2f1b49314976a8bfce65 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Thu, 24 Feb 2022 11:20:06 +0100
Subject: [PATCH] Add DBus service file to actually make it activatable

org.kde.amarok.desktop specifies DBusActivatable=true, but that only worked
if amarok was already running. Install a service file to make it actually
activatable.
---
 src/CMakeLists.txt                 | 3 +++
 src/dbus/org.kde.amarok.service.in | 3 +++
 2 files changed, 6 insertions(+)
 create mode 100644 src/dbus/org.kde.amarok.service.in

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a70135f970..cbb3f8f542 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -952,6 +952,9 @@ install(FILES dbus/org.kde.amarok.App.xml
               dbus/org.kde.amarok.Collection.xml
               dbus/mpris2/org.kde.amarok.Mpris2Extensions.Player.xml
               DESTINATION ${KDE_INSTALL_DBUSINTERFACEDIR})
+configure_file(dbus/org.kde.amarok.service.in ${CMAKE_CURRENT_BINARY_DIR}/dbus/org.kde.amarok.service @ONLY)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/dbus/org.kde.amarok.service
+              DESTINATION ${KDE_INSTALL_DBUSSERVICEDIR})
 
 install(FILES services/InfoParserLoading.html
               browsers/hover_info_template.html
diff --git a/src/dbus/org.kde.amarok.service.in b/src/dbus/org.kde.amarok.service.in
new file mode 100644
index 0000000000..25953cedc2
--- /dev/null
+++ b/src/dbus/org.kde.amarok.service.in
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=org.kde.amarok
+Exec=@KDE_INSTALL_FULL_BINDIR@/amarok
-- 
GitLab

